jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java

```diff
@@ -513,22 +519,28 @@ protected Client getWSClient(WorkItem workItem, String interfaceRef) {
 	                                                                                  interfaceRef),
 	                                                                        getInternalClassLoader(),
 	                                                                        null);
-	                        setClientTimeout(workItem, client);
-	                        clients.put(interfaceRef,
-	                                    client);
-	                        return client;
-	                    } catch (Exception e) {
-	                        logger.error("Error when creating WS Client",
-	                                     e);
-	                        continue;
-	                    }
-	                }
-	            }
-	        }
+                            setClientTimeout(workItem, client);
+                            Collection<WorkItemHeaderInfo> headers = WorkItemHeaderUtils.getHeaderInfo(workItem);
+                            if (!headers.isEmpty()) {
+                                client.getRequestContext().put(Header.HEADER_LIST, headers.stream().map(
+                                        this::buildHeader).collect(Collectors.toList()));
+                            }
+                            clients.put(interfaceRef, client);
+                            return client;
+                        } catch (Exception e) {
+                            logger.error("Error when creating WS Client", e);
+                        }
+                    }
+                }
+            }
         }
         return null;
     }
 
+    private Header buildHeader(WorkItemHeaderInfo header) {
+        return new Header(new QName(header.getParam("NS").toString(), header.getName()), header.getContent());
```

Comment:
minor comment here if the NS is mandatory ?

Reply:
no, thanks for pointing out, Ill change it